### PR TITLE
fix nil pointer panic when changing bucket config

### PR DESCRIPTION
**Pull request by degorenko**
Related-Issue: https://github.com/kube-object-storage/lib-bucket-provisioner/issues/219

Signed-off-by: Denis Egorenko <degorenko@mirantis.com>

### CODE REVIEW

File: pkg/provisioner/controller.go [hunk 1]
```diff
@@ -707,10 +707,14 @@ func (c *obcController) handleUpdateClaim(key string, obc *v1alpha1.ObjectBucket
...
-		destAdditionalConfig[key] = value
+	if destAdditionalConfig != nil {
+		for key, _ := range destAdditionalConfig {
+			delete(destAdditionalConfig, key)
+		}
+		for key, value := range srcAdditionalConfig {
+			destAdditionalConfig[key] = value
+		}
+	} else {
+		destAdditionalConfig = srcAdditionalConfig
```
Comment (BlaineEXE) [inline pkg/provisioner/controller.go:718]: I'm not sure if this is the source of the issue, but the copy must be a deep copy.

```suggestion
	if destAdditionalConfig != nil {
		for key, _ := range destAdditionalConfig {
			delete(destAdditionalConfig, key)
		}
	}
	for key, value := range srcAdditionalConfig {
		destAdditionalConfig[key] = value
	}
```

**Review by BlaineEXE — CHANGES_REQUESTED** [review]

**Comment by BlaineEXE** [comment]
I think this looks good. Would you describe the manual tests you used to test that the fix is working? I'd like to make sure there aren't any corner cases missed

**Comment by degorenko** [comment]
@BlaineEXE  so, i did fork, made changes. But i'm currently using Rook 1.8.5, so i have to rebase on commit which is used in go.mod for 1.8.5 version. Finally, i have v0.0.0-20220105185820-c1da9586e05b plus that changes on top of it (I have additional config-override-1.8 branch in fork). Replaced in Rook go.mod l-b-p dependency on my fork commit, built Rook image on top of it and then test. So in my tests Rook 1.8.5+current change. Does it suites your expectations ?

**Comment by BlaineEXE** [comment]
I was thinking to look at the logs and make sure...
- create OBC successful
- no changes to OBC (restart operator), nothing happens
- change OBC additional config, changes happen correctly now
- change another OBC field, controller errors saying it can't be updated
- delete OBC works

**Comment by degorenko** [comment]
[rook-operator.log](https://github.com/kube-object-storage/lib-bucket-provisioner/files/9310460/rook-operator.log)
[user-stats.log](https://github.com/kube-object-storage/lib-bucket-provisioner/files/9310462/user-stats.log)

**Comment by degorenko** [comment]
log contains some debug message around OBC spec diff, but it is not in current change, only for test purpose.

**Comment by BlaineEXE** [comment]
The tests look great. Thank you. I'm trying to figure out why the CI is failing though.

**Comment by BlaineEXE** [comment]
It looks like one of the tools used in CI is no longer hosted at the location. 

This line in travis.yml needs updated: https://github.com/kube-object-storage/lib-bucket-provisioner/blob/4bea5edaff58c0e80e3ad0c56f2894a7d456cc87/.travis.yml#L18 

Replacing that with this should get it going again I think.
```
curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/v1.16.0/install.sh | sh -s -- -b $(go env GOPATH)/bin && golangci-lint --version
```

**Review by BlaineEXE — APPROVED** [review]

**Comment by BlaineEXE** [comment]
Merged. To pull the change into Rook, we just need a PR that updates the go.mod to use the latest version (hash) of this library. And we can have it backported back to 1.8 and 1.9 as well. 👍 Thanks again!
